### PR TITLE
ImaVid with new timeline API

### PR DESCRIPTION
**Pull request by sashankaryal**
## What changes are proposed in this pull request?

Rewrite ImaVid with new Timeline API.

Todo:
- [x] center frame number div
- [x] implement speed
- [x] pause and play on space
- [x] allow animation loop to be controlled by creator
- [x] show buffering status (multi-range)
- [x] update LOAD_RANGE_SIZE
- [x] labels / detections should pause fetching when animation loop is on
- [x] fix failing e2es

## How is this patch tested? If it is not, please explain why.

E2E to catch regressions.

## Release Notes

- **New Features**
  - Introduced the `ImaVidLookerReact` component for enhanced video viewing with a timeline feature.
  - Added the `FrameLoaderView` component for managing frame loading within a timeline context.

- **Improvements**
  - Streamlined the `ModalLooker` component structure for better rendering logic.
  - Enhanced playback controls by focusing exclusively on `VideoState`.
  - Updated element handling in the `getImaVidElements` function to improve dynamic rendering based on thumbnail state.

- **Bug Fixes**
  - Streamlined event handling for video frame controls and navigation.

- **Styles**
  - Added new CSS classes for improved styling of video controls and seekbars, including `.imaVidLookerControls`, `.imaVidSeekBar`, and `.hideInputThumb`.

### CODE REVIEW

File: app/packages/state/src/hooks/hooks-utils.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const useScrollHandler = (handler) =>
 export const useHashChangeHandler = (handler) =>
   useEventHandler(window, "hashchange", handler);
 
-export const useKeydownHandler = (handler: React.KeyboardEventHandler) =>
+export const useKeydownHandler = (handler: (e: KeyboardEvent) => void) =>
```
Comment (benjaminpkane) [inline app/packages/state/src/hooks/hooks-utils.ts:69]: 🥇

File: app/packages/core/src/components/Modal/ImaVidLooker.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+      looker.attach(id);
+    }, [looker, id]);
+
+    useEventHandler(looker, "clear", useClearSelectedLabels());
+
+    const hoveredSample = useRecoilValue(fos.hoveredSample);
+
+    useEffect(() => {
+      const hoveredSampleId = hoveredSample?._id;
+      looker.updater((state) => ({
```
[the remark this answers is not in the document]
Comment (benjaminpkane) [reply app/packages/core/src/components/Modal/ImaVidLooker.tsx:138]: I agree with 🐰 here. I'm not sure this hook is necessary

**Review by ritch — APPROVED** [review]
This works as expected. I've tested it via the python timeline panel.

**Review by benjaminpkane — APPROVED** [review]
LGTM 🚀 (assuming passing e2e)

**Pull request merged**